### ISSUE TITLE
ZIO `andThen` is broken 

### ISSUE THREAD

**Issue by susliko** [zio/zio#3296]
The method `.andThen` is now defined as 
```scala
final def andThen[R1 >: A, E1 >: E, B](that: ZIO[R1, E1, B]): ZIO[R, E1, B]
```
While its context type parameter bound should be `R1 >: R` instead of `R1 >: A`

**Comment by adamgfraser** [comment]
I don't think it is broken but the type signature can be simplified. This method is supposed to take the output type of this effect, the `A` value, and send it to a second effect that requires the `A` as its environment type, representing arrow composition.

The signature can be simplified as done in [this](https://github.com/zio/zio/pull/3294/files) open PR to:

```scala
final def andThen[E1 >: E, B](that: ZIO[A, E1, B]): ZIO[R, E1, B] =
  self >>> that
```

**Comment by adamgfraser** [comment]
Closed by #3294.
